test/integration/redirect_integration_test.cc

```diff
@@ -197,6 +197,67 @@ TEST_P(RedirectIntegrationTest, BasicInternalRedirect) {
   EXPECT_THAT(waitForAccessLog(access_log_name_, 1), HasSubstr("200 via_upstream -"));
 }
 
+// Test that Envoy should correctly respond to "Connection: close" header in requests which get
+// redirected by echoing "Connection: close" in response and closing the connection immediately.
+TEST_P(RedirectIntegrationTest, ConnectionCloseHeaderHonoredInInternalRedirect) {
+  config_helper_.addRuntimeOverride(
+      "envoy.reloadable_features.http1_connection_close_header_in_redirect", "true");
```

Comment:
Is there any test cover the case of `false`? It will be great to have one

Reply:
I thought even if it isn't wrong behavior, it doesn't mean there is no-one depend on it. that the point of we need runtime flag here, that is for backward-compatible and deprecate the old behavior in the future.  Then we need ensure the old behavior still works for people still depends on it. 

After we remove the old behavior and runtime flag, those tests can be removed also.

Anyway, thanks for update it.